### PR TITLE
Fix for slice OneDNN kernel in solov2 and ppyolo models

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
Bug fixes

### PR changes
OPs 

### Describe
Fix for slice op which was causing error in solov2 model inference. Error was mentioned in #34723

### CODE REVIEW

**Review by jczaja — DISMISSED** [review]
LGTM

**Comment by jczaja** [comment]
@tsocha Please review this PR

File: paddle/fluid/operators/mkldnn/slice_mkldnn_op.cc [hunk 1]
```diff
@@ -79,11 +95,27 @@ class SliceMKLDNNKernel : public framework::OpKernel<T> {
...
+      slice_dims[decrease_axis[i]] = 0;
+    }
+
+    int j = 0;
+    for (size_t i = 0; i < slice_dims.size(); ++i) {
+      if (slice_dims[i] != 0) new_out_dims[j++] = slice_dims[i];
+    }
+
+    // paddle doesn't support rank 0 tensors, instead we use [1]
+    if (new_out_dims.size() == 0) new_out_dims.emplace_back(1);
```
Comment (tsocha) [inline paddle/fluid/operators/mkldnn/slice_mkldnn_op.cc]: You can use a range based for loop and skip computation for empty output dims:
```suggestion
    std::vector<int64_t> new_out_dims(slice_dims.size() - decrease_axis.size());

    // paddle doesn't support rank 0 tensors, instead we use [1]
    if (new_out_dims.size() == 0) new_out_dims.emplace_back(1);
    else {
        for (const auto& axis: decrease_axis) {
          slice_dims.at(axis) = 0;
        }

        int i = 0;
        for (const auto& slice_dim: slice_dims) {
          if (slice_dim != 0) new_out_dims.at(i++) = slice_dim;
        }
    }
```

**Comment by jakpiase** [comment]
Added a fix for #35717

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Pull request merged**